File: mifosng-android/src/main/java/com/mifos/api/local/databasehelper/DatabaseHelperStaff.java
package com.mifos.api.local.databasehelper;

import android.os.AsyncTask;

import com.mifos.objects.organisation.Staff;
import com.mifos.objects.organisation.Staff_Table;
import com.raizlabs.android.dbflow.sql.language.SQLite;

import org.reactivestreams.Subscriber;

import java.util.List;
import java.util.concurrent.Callable;

import javax.inject.Inject;
import javax.inject.Singleton;

import io.reactivex.Observable;
import io.reactivex.ObservableSource;


/**
 * Created by Rajan Maurya on 7/7/16.
 */
@Singleton
public class DatabaseHelperStaff {


    @Inject
    public DatabaseHelperStaff() {

    }


    public Observable<Void> saveAllStaffOfOffices(final List<Staff> staffs) {
        AsyncTask.THREAD_POOL_EXECUTOR.execute(new Runnable() {
            @Override
            public void run() {

                for (Staff staff : staffs) {
                    staff.save();
                }

            }
        });

        return null;
    }


    public Observable<List<Staff>> readAllStaffOffices(final int officeId) {
        return Observable.defer(new Callable<ObservableSource<? extends List<Staff>>>() {
            @Override
            public ObservableSource<? extends List<Staff>> call() throws Exception {
                List<Staff> staffs = SQLite.select()
                        .from(Staff.class)
                        .where(Staff_Table.officeId.eq(officeId))
                        .queryList();

                return Observable.just(staffs);
            }
        });
    }
}

File: core/datastore/src/main/java/com/mifos/core/databasehelper/DatabaseHelperStaff.kt
package com.mifos.core.databasehelper

import android.os.AsyncTask
import com.mifos.core.objects.organisation.Staff
import com.mifos.core.objects.organisation.Staff_Table
import com.raizlabs.android.dbflow.sql.language.SQLite
import rx.Observable
import javax.inject.Inject
import javax.inject.Singleton

/**
 * Created by Rajan Maurya on 7/7/16.
 */
@Singleton
class DatabaseHelperStaff @Inject constructor() {
    fun saveAllStaffOfOffices(staffs: List<Staff>): Observable<Void>? {
        AsyncTask.THREAD_POOL_EXECUTOR.execute {
            for (staff in staffs) {
                staff.save()
            }
        }
        return null
    }

    fun readAllStaffOffices(officeId: Int): Observable<List<Staff>> {
        return Observable.create { subscriber ->
            val staffs = SQLite.select()
                .from(Staff::class.java)
                .where(Staff_Table.officeId.eq(officeId))
                .queryList()
            subscriber.onNext(staffs)
        }
    }
}